src/audio/google/google_rtc_audio_processing.c

```diff
@@ -484,44 +515,55 @@ static void google_rtc_audio_processing_free(struct comp_dev *dev)
 	rfree(cd->raw_mic_buffer);
 	comp_data_blob_handler_free(cd->tuning_handler);
 	rfree(cd);
-	rfree(dev);
-}
-
-
-static int google_rtc_audio_processing_trigger(struct comp_dev *dev, int cmd)
-{
-	comp_dbg(dev, "google_rtc_audio_processing_trigger(): %d", cmd);
-
-	return comp_set_state(dev, cmd);
+	return 0;
 }
 
-static int google_rtc_audio_processing_prepare(struct comp_dev *dev)
+static int google_rtc_audio_processing_prepare(struct processing_module *mod,
+					       struct sof_source **sources,
+					       int num_of_sources,
+					       struct sof_sink **sinks,
+					       int num_of_sinks)
 {
-	struct google_rtc_audio_processing_comp_data *cd = comp_get_drvdata(dev);
+	struct comp_dev *dev = mod->dev;
+	struct google_rtc_audio_processing_comp_data *cd = module_get_private_data(mod);
 	struct list_item *source_buffer_list_item;
+	struct comp_buffer *output;
 	unsigned int aec_channels = 0, frame_fmt, rate;
+	int microphone_stream_channels = 0;
+	int output_stream_channels;
 	int ret;
+	int i = 0;
+
+	comp_info(dev, "google_rtc_audio_processing_prepare()");
 
-	comp_dbg(dev, "google_rtc_audio_processing_prepare()");
+#if CONFIG_IPC_MAJOR_4
+	google_rtc_audio_processing_params(mod);
+#endif
 
 	/* searching for stream and feedback source buffers */
 	list_for_item(source_buffer_list_item, &dev->bsource_list) {
 		struct comp_buffer *source = container_of(source_buffer_list_item,
 							  struct comp_buffer, sink_list);
-
 #if CONFIG_IPC_MAJOR_4
 		if (IPC4_SINK_QUEUE_ID(source->id) == SOF_AEC_FEEDBACK_QUEUE_ID) {
 #else
 		if (source->source->pipeline->pipeline_id != dev->pipeline->pipeline_id) {
 #endif
-			cd->aec_reference = source;
```

Comment:
It's because module adapter provides the input and ouput streams as arrays of pointers to process(). I use the index to access the right stream. This also eases the transition to source/sink api in pipeline2.0 with better abstraction where process() receives array of sof_source and sof_sink pointers. It let's us optimize further the pipeline buffers. I'm not sure keeping the old way would prevent that but I felt this is cleaner approach.

What do you think @marcinszkudlinski , was this the right thing to do?